pyqt-apps/siriushla/as_ps_cycle/util.py

```diff
@@ -0,0 +1,207 @@
+"""Utilities for ps-cycle."""
+
+import time as _time
+import epics as _epics
+from math import isclose as _isclose
+
+from siriuspy.envars import vaca_prefix as VACA_PREFIX
+from siriuspy.search.ma_search import MASearch
+from siriuspy.csdevice.pwrsupply import Const as _PSConst
+
+
+CONNECTION_TIMEOUT = 0.05
+
+
+def get_manames():
+    """."""
+    return MASearch.get_manames({'dis': 'MA'})
+
+
+class Timing:
+    """Timing."""
+
+    DEFAULT_DURATION = 150  # [us]
+    DEFAULT_NRPULSES = 1
+    DEFAULT_DELAY = 0  # [us]
+    DEFAULT_POLARITY = 'Normal'  # test
+
+    _properties = {
+
+        'RA-RaMO:TI-EVG:CycleMode-Sel': 'External',
+        'RA-RaMO:TI-EVG:UpdateEvt-Cmd': 1,
+        'RA-RaMO:TI-EVG:CycleExtTrig-Cmd': None,  # There is not default value
+
+        'BO-Glob:TI-Mags:Src-Sel': 'Cycle',
+        'BO-Glob:TI-Mags:NrPulses-SP': DEFAULT_NRPULSES,
+        'BO-Glob:TI-Mags:Duration-SP': DEFAULT_DURATION,
+        'BO-Glob:TI-Mags:Delay-SP': DEFAULT_DELAY,
+        'BO-Glob:TI-Mags:Polarity-Sel': DEFAULT_POLARITY,  # test
+        'BO-Glob:TI-Mags:State-Sel': 'Enbl',
+
+        'BO-Glob:TI-Mags:Src-Sel': 'Cycle',
+        'BO-Glob:TI-Mags:NrPulses-SP': DEFAULT_NRPULSES,
+        'BO-Glob:TI-Mags:Duration-SP': DEFAULT_DURATION,
+        'BO-Glob:TI-Mags:Delay-SP': DEFAULT_DELAY,
+        'BO-Glob:TI-Mags:Polarity-Sel': DEFAULT_POLARITY,  # test
+        'BO-Glob:TI-Mags:State-Sel': 'Enbl',
```

Comment:
@xresende, I think there's a mistake here... you are defining default values to the same trigger again.

Reply:
fixed and added other TI trigger PVs (BO-Corr and TB-Mags)